### PR TITLE
[WIP] optimize dataloader for cpu path aand enable topk bf16 type

### PR DESCRIPTION
**Pull request by XiaobingSuper**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#37331 [WIP] optimize dataloader for cpu path aand enable topk bf16 type**

### CODE REVIEW

File: torch/utils/data/dataloader.py [hunk 1]
```diff
@@ -9,11 +9,13 @@
 import itertools
 import warnings
 
+
 import multiprocessing as python_multiprocessing
 import torch
 import torch.multiprocessing as multiprocessing
 from torch._utils import ExceptionWrapper
 from torch._six import queue, string_classes
+import psutil
```
Comment (vadimkantorov) [inline torch/utils/data/dataloader.py:18]: Is `psutil` a new dependency of PyTorch?

Comment (XiaobingSuper) [reply torch/utils/data/dataloader.py:18]: This is just for test.

**Comment by ezyang** [comment]
Remove WIP from title when you need to find a reviewer

File: torch/utils/data/_utils/worker.py [hunk 2]
```diff
@@ -161,6 +162,8 @@ def _worker_loop(dataset_kind, dataset, index_queue, data_queue, done_event,
         iteration_end = False
 
         watchdog = ManagerWatchdog()
+        p = psutil.Process()
```
Comment (vadimkantorov) [inline torch/utils/data/_utils/worker.py:165]: Should `psutil` be added to the list of required PyTorch dependencies? At the moment, it seems it's not required and various checks are made in the codebase to check if it's installed:
https://github.com/pytorch/pytorch/blob/21c38e17997171415a44c6ba578f621037d8ef30/test/test_dataloader.py

**Pull request closed without merging**